### ISSUE TITLE
Linker Script Question

### ISSUE THREAD

**Issue by zgibson21** [akospasztor/stm32-bootloader#40]
Hello, I am trying to port this to work on a STM32F107 based custom board and was hoping for some clarification on how to setup the .ld file. From your example I see that the start address is mapped to 8000000 with a length of 8000.

Do you also have to setup a flash partition for the application memory? It seems like the code is able to jump to the main application at the 80008000 address but that memory region is not defined in the .ld file.

My end goal is to have it set up with the Bootloader, app memory, an update cache (for storing downloaded bin files) and a small region at the end for storing some user data.

Could you please clarify if the .ld file needs this information or not?

**Comment by akospasztor** [comment]
Hi there,

In your bootloader project, you should set up the .ld file (or .sct if you're using Keil, or .icf if you're using IAR) so that it reflects only the bootloader image only and does not include the application image (or any other region that is not used by the bootloader). The bootloader and the application should know ideally "nothing" about the existence of each other. Specifically, the bootloader should not know anything about the application itself (except where to jump, which is the application starting address).

Within the application project, the only thing that you should adjust is the starting address of the application. This means you should configure your .ld file so that the starting address is the actual address (within your MCU flash) where your application starts. Additionally, either of the bootloader or the application itself should set the vector table location of the application so that the interrupts within your application would work. Please refer to the README (https://github.com/akospasztor/stm32-bootloader#how-to-use) for detailed description. 

In conclusion, you should perform the followings:
1. Configure the .ld file of the bootloader to reflect only the bootloader region in your flash (e.g. starting from 0x0800 0000 with a size of 0x8000). 
2. Configure the .ld file of your application so that it reflects your application region (e.g. starting from 0x0800 8000 with a size of e.g. 0x10000).
3. For the update cache and your configuration region, you can reserve an empty region within your application's .ld file, but it is actually not necessary.

I hope this helps.

Best,
Akos
